changesets/scripts/entrypoint.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+search_key=""
+search_value=""
+cached=false
+modify_string="hgv=no to hgv=yes"
+
+escape_spaces() {
+  echo "${1// /\\ }"
+}
+
+USAGE_STRING="Correct usage: generate_changeset.sh --search-key \"<search key>\" --search-value \"<search value>\""
+OPTSTRING="k:v:c"
+LONGOPTSTRING="search-key:,search-value:,cached"
+
+if ! ARGS=$(getopt -o "$OPTSTRING" --long "$LONGOPTSTRING" -- "$@"); then
+  echo "Failed to parse args"
+  echo "$USAGE_STRING"
+  exit 1
+fi
+
+eval set -- "$ARGS"
```

Comment:
question: why `eval set` and not `set`?

```suggestion
set -- "$ARGS"
```

Reply:
https://manpages.debian.org/bullseye/util-linux/getopt.1.en.html#:~:text=Traditional%20implementations%20of,can%20be%20used.